### ISSUE TITLE
Text Report Misses Last Class

### ISSUE THREAD

**Issue by spdavern** [Chien-Hung/DetVisGUI#6]
This is a minor issue, but the text report that DetVisGUI.py produces during initialization misses the last class in the list.  For example when I run DetVisGUI.py on my custom dataset that has 3 object classes (as shown by "Total Category") I get:
```
==============[ COCO json info ]==============
Total Annotations: 216
Total Image      : 172
Annotated Image  : 172
Total Category   : 3
----------------------------
       class        | count
----------------------------
       human        | 86
      vehicle       | 130
```
That result is missing the animal class in my case.
The source of the error is within the parse_jason function, line 112 of DetVisGUI.py in the mmDetection branch:
`112        category_count = [0 for _ in range(max_category_id)]`
Corrected to:
`112        category_count = [0 for _ in range(max_category_id + 1)]`
Shows all the classes and their counts:
```
==============[ COCO json info ]==============
Total Annotations: 216
Total Image      : 172
Annotated Image  : 172
Total Category   : 3
----------------------------
       class        | count
----------------------------
       human        | 86
      vehicle       | 59
       animal       | 71
```

Thanks for the cool tool!

**Comment by Chien-Hung** [comment]
Hello, this error is caused by line 142 ~ 144 : 
``` python
for c, cnt in zip(category, category_count):
    if cnt != 0:
        print('{:^20}| {}'.format(c, cnt))
```
I fixed this error by change these lines to 
``` python
for idx, cat_idx in enumerate(cat2idx): 
    c = category[idx]
    cnt = category_count[cat_idx-1]
    if cnt != 0:
        print('{:^20}| {}'.format(c, cnt))   
```
You can test it on your dataset and if this issue is solved, please close this issue. If there are any errors, please let me know. Thank you.

**Comment by spdavern** [comment]
Hello, I removed the fix I suggested and instead made the change you suggested.  This gives incorrect counts because the length of the category_count list is then 1 short:
```
==============[ COCO json info ]==============
Total Annotations: 216
Total Image      : 172
Annotated Image  : 172
Total Category   : 3
----------------------------
       class        | count
----------------------------
       human        | 130
      vehicle       | 86
       animal       | 130
```
However, you're right the loop at 142~144 also needs to be changed because my suggested fix leaves the wrong count associated with each class. If, however, we make BOTH of our changes we get the right result. Here is the correct count per class result which I verified independently:
```
==============[ COCO json info ]==============
Total Annotations: 216
Total Image      : 172
Annotated Image  : 172
Total Category   : 3
----------------------------
       class        | count
----------------------------
       human        | 71
      vehicle       | 86
       animal       | 59
```
I'm so glad you checked that because I neglected to.  I previously saw the right set of values but didn't check that they were associated with the right class.  Now, they're right!

**Comment by Chien-Hung** [comment]
Hi, I think this error is caused by the start number of category id in the json annotation. COCO category id is started from 1, and I guess your category id is started from 0. I revise some [code](https://github.com/Chien-Hung/DetVisGUI/commit/881105a0cf298bf6a62471aed9839b089feea9da) based on your suggestion which can make 2 case running correct. Please help check again, thank you. :smiley:

**Comment by spdavern** [comment]
Thank you!  Yes, I pulled the updates and confirmed that this appears to be fixed.  I'll close this issue.

Out of curiosity, I couldn't find guidance to definitively support that the category id should be 1-based.  I checked the [COCO Data Format](https://cocodataset.org/#format-data) page, the [COCO API](https://github.com/cocodataset/cocoapi) repo including the [coco.py](https://github.com/cocodataset/cocoapi/blob/master/PythonAPI/pycocotools/coco.py) script.  I did confirm that at least for instances_val2017.json, from the official COCO dataset, the "categories" list id is 1 based.  However, I'm not sure that is common or universal among annotations for a wide range of datasets using the COCO annotation format.  I've previously wondered how problematic it might be for any of the annotation lists to not even be contiguous, much less 0 or 1-based.  Anyway, if you know of a more clear source for that convention I'd be happy to know it too.  Regardless, clearly your tool is more robust to not depend on the 0/1 basis.  Thank you for providing it.
